### PR TITLE
Recheck if CPU is balanced if cpufreq stays constant

### PR DESCRIPTION
**Pull request by djselbeck**
This fixes an edge case where no re-evaluation of CPU balance
is performed. This could happen with the following situation:

3 cores online, core_0 and core_1 at around 100 load and max cpufreq.
 * core_3 clocks down below idle_bottom_freq
 * rqbalance_work_func will result in shutdown of core_3
 * repeat: rqbalance_work_func stays in rqbalance_state == DOWN
         => No re-evaluation of CPU load balance until load
            is reduced and therefore cpufreq changes

### CODE REVIEW

**Comment by Bitti09** [comment]
@djselbeck I'll test and report tomorrow

_Sent from my Xperia XZ (AOSP) using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by kholk** [comment]
Makes some sense, but needs careful testing.

**Comment by Bitti09** [comment]
first short test with   360° youtube and pinout game : much less stuttering than before.

**Comment by alviteri** [comment]
@kholk first test pass on dora seeing galeazzi's videos on 1080p60fps on youtube ..... device didn't freeze on android oreo ...... will report other things

**Comment by kholk** [comment]
That's good guys, thank you!
Now please test this on a daily driver device and please compare heat and power consumption.

I will test it in 1 to 2 days. Please test for at least 2 days.

**Comment by alviteri** [comment]
Mobile Legends game ok on tone

**Comment by djselbeck** [comment]
I've been running it on my daily driver since opening this PR on kagura. 

Also I've tested if it powers down to 1 core which happens but very rarely because the core 0 and 1 are tied together by cpufreq and the cpufreq based rqbalancing kicks in often.

**Comment by kholk** [comment]
Don't misunderstand me, the logic IMHO is completely fine and the change is also good.

Being this a very important driver, common between *all* devices, I am paranoid about anything that can happen due to this change. This is why I'm asking you all to thoroughly test it.

**Comment by djselbeck** [comment]
Of course, I can completely understand this ;).

**Comment by alviteri** [comment]
Probably is not related but poplar is a disaster with this .... It lags as hell .... Need more test

**Comment by Bitti09** [comment]
@alviteri for me there are no lags with this PR

_Sent from my Xperia XZ (AOSP) using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by alviteri** [comment]
@Bitti09 as angelo said needs to be tested on all

**Comment by kholk** [comment]
Interesting. The change actually tends to increase the computational power, not to decrease it, as it makes the governor more reactive to upcore decision. 

Please be sure that the lags are related to this pr.

File: drivers/cpuquiet/governors/rqbalance.c [hunk 1]
```diff
@@ -555,13 +555,18 @@ static void rqbalance_work_func(struct work_struct *work)
...
+			rqbalance_state = UP;
+		}
 		else
 			stop_load_timer();
 
-			queue_delayed_work(rqbalance_wq,
-						 &rqbalance_work, up_delay);
+
+
+		queue_delayed_work(rqbalance_wq,
```
Comment (alviteri) [inline drivers/cpuquiet/governors/rqbalance.c]: @djselbeck can you update this ..... string 558 - 562 are the commit string 566-569 is useless

Comment (kholk) [reply drivers/cpuquiet/governors/rqbalance.c]: I agree about the useless whitelines

Comment (djselbeck) [reply drivers/cpuquiet/governors/rqbalance.c]: Yes the lines should be fixed, sadly I noticed them to late. But the indentation should be fixed I guess?

**Comment by oshmoun** [comment]
would it make sense to make the recheck happen less frequently?
would something like this pseudo code work?
```
if (cpu < nr_cpu_ids) {
    if(all_online_cores_are_at_max_freq()) {
        // force a recheck because cpufreq has reached its upper limit
         rqbalance_state = UP;
   }
}
```
or would such code produce even more overhead than forcing the recheck every time?

File: drivers/cpuquiet/governors/rqbalance.c (hunk 1 above)
Comment (kholk) [reply drivers/cpuquiet/governors/rqbalance.c]: I don't see any indentation problem

**Comment by kholk** [comment]
What you propose introduces unnecessary overhead.......
Probably we should just restore rqbalance's functionality as per my original design.

File: drivers/cpuquiet/governors/rqbalance.c (hunk 1 above)
Comment (alviteri) [reply drivers/cpuquiet/governors/rqbalance.c]: you should rebase ... this is useless commit

**Comment by alviteri** [comment]
https://github.com/SonyAosp/kernel/commit/6a91672dbfb2e401dde3aac87e190ec12d1b9493
https://github.com/SonyAosp/kernel/commit/f4b9b6561d7255eb9e2b61771c35aaebe7fbf94f

**Comment by djselbeck** [comment]
The code should be ok now?

**Comment by Bitti09** [comment]
any updates on this ?

**Comment by RyTekk** [comment]
It should be closed, doesn't it?

**Comment by kholk** [comment]
Yep. Another solution was applied.

**Comment by djselbeck** [comment]
Yep, close this. Although I can still trigger the old behaviour on kagura. If I've time i'll check if I can find the issue with the new solution.

**Pull request closed without merging**